src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -324,6 +322,20 @@ public List<DarCollectionSummary> getSummariesForRoleName(User user, String user
     return summaries;
   }
 
+  private List<Integer> getDatasetIdsForUserAndRole(User user, UserRoles role) {
+    List<Integer> roleDacIds = user.getRoles().stream()
+        .filter(ur -> ur.getRoleId().equals(role.getRoleId()))
+        .map(UserRole::getDacId)
+        .filter(Objects::nonNull)
+        .toList();
+    return Stream.of(roleDacIds)
```

Comment:
I'm curious about the choice of a flatMap here.  Seems like this could all be simplified by having `datasetDAO::findDatasetListByDacIds` just return an empty list if the input is an empty list no?

Reply:
Our DAO calls are just interface methods (jdbi) that just execute a simple statement. If the list is empty, we get a sql error, so I tend to err on the side of safety and prevent that in most places.

Reply:
Yeah, my first inclination was to make the DAO layer check for an empty list. But since you have an auto-generated DAO it makes more sense to address it in the caller, as long as there's only one caller.